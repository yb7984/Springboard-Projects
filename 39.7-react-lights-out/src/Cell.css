.Cell {
  height: 50px;
  width: 50px;
  background-color: darkgray;
  cursor: pointer;
  border-radius: 10px;
}
.Cell:hover{
  background-color: lightgray;
}

.Cell-lit {
  background-color: white;
  transition: all 1s;
}