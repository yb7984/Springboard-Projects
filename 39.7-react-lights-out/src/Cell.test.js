import React from "react";
import { render, asFragment } from "@testing-library/react";
import Cell from "./Cell";

it("renders without crashing", function () {
    render(<table><tbody><tr><Cell /></tr></tbody></table>);
});

it("matches snapshot", function () {
    const { asFragment } = render(<table><tbody><tr><Cell /></tr></tbody></table>);
    expect(asFragment()).toMatchSnapshot();
});


it("works when light is on", function () {
    const { asFragment, queryByTestId } = render(<table><tbody><tr><Cell isLit={true} cellId="1-1" /></tr></tbody></table>);
    
    expect(queryByTestId("1-1")).toHaveClass("Cell-lit");
});


it("works when light is off", function () {
    const { asFragment, queryByTestId } = render(<table><tbody><tr><Cell isLit={false} /></tr></tbody></table>);
    
    expect(queryByTestId("cell")).not.toHaveClass("Cell-lit");
});
