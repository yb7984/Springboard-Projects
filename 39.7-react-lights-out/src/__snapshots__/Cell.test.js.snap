// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell "
          data-testid="cell"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
